File: transektcount/src/main/java/com/wmstein/transektcount/widgets/CountingWidget_head3.java
/*
 * Copyright (c) 2016. Wilhelm Stein, Bonn, Germany.
 */

package com.wmstein.transektcount.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.wmstein.transektcount.R;

import java.util.Objects;

/****************************************************
 * Interface for widget_counting_head3.xml
 * Created by wmstein 18.12.2016
 * Last edited on 2020-04-17
 */
public class CountingWidget_head3 extends RelativeLayout
{
    private TextView countHead3;

    public CountingWidget_head3(Context context, AttributeSet attrs)
    {
        super(context, attrs);

        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        Objects.requireNonNull(inflater).inflate(R.layout.widget_counting_head3, this, true);
        countHead3 = findViewById(R.id.countHead3);
    }

    public void setCountHead3()
    {
        countHead3.setText(getContext().getString(R.string.countExternalHint));
    }

}

File: transektcount/src/main/java/com/wmstein/transektcount/widgets/CountingWidget_head3.kt
/*
 * Copyright (c) 2016. Wilhelm Stein, Bonn, Germany.
 */
package com.wmstein.transektcount.widgets

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.RelativeLayout
import android.widget.TextView
import com.wmstein.transektcount.R
import java.util.Objects

/****************************************************
 * Interface for widget_counting_head3.xml
 * Created by wmstein 18.12.2016
 * Last edited in Java on 2021-01-26
 * converted to Kotlin on 2023-06-26
 */
class CountingWidget_head3(context: Context, attrs: AttributeSet?) :
    RelativeLayout(context, attrs) {
    private val countHead3: TextView

    init {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        Objects.requireNonNull(inflater).inflate(R.layout.widget_counting_head3, this, true)
        countHead3 = findViewById(R.id.countHead3)
    }

    fun setCountHead3() {
        countHead3.text = context.getString(R.string.countExternalHint)
    }
}